Core/src/org/sleuthkit/autopsy/contentviewers/MediaPlayerPanel.java

```diff
@@ -531,22 +549,236 @@ protected void done() {
+            Rectangle thumb = this.thumbRect;
+
+            Color original = graphic.getColor();
+
+            //Change the thumb view from the rectangle
+            //controller to an oval.
+            graphic.setColor(config.getThumbColor());
+            Dimension thumbDimension = config.getThumbDimension();
+            graphic.fillOval(thumb.x, thumb.y, thumbDimension.width, thumbDimension.height);
+
+            //Preserve the graphics original color
+            graphic.setColor(original);
+        }
+
+        @Override
+        public void paintTrack(Graphics graphic) {
+            //This rectangle is the bounding box for the progress bar
+            //portion of the slider. The track is painted in the middle
+            //of this rectangle and the thumb laid overtop.
+            Rectangle track = this.trackRect;
+
+            //Get the location of the thumb, this point splits the
+            //progress bar into 2 line segments, seen and unseen.
+            Rectangle thumb = this.thumbRect;
+            int thumbX = thumb.x;
+            int thumbY = thumb.y;
+
+            Color original = graphic.getColor();
+
+            //Paint the seen side
+            graphic.setColor(config.getSeenTrackColor());
+            graphic.drawLine(track.x, track.y + track.height / 2,
+                    thumbX, thumbY + track.height / 2);
+
+            //Paint the unseen side
+            graphic.setColor(config.getUnseenTrackColor());
+            graphic.drawLine(thumbX, thumbY + track.height / 2,
+                    track.x + track.width, track.y + track.height / 2);
+
+            //Preserve the graphics color.
+            graphic.setColor(original);
+        }
+
+        @Override
+        protected TrackListener createTrackListener(JSlider slider) {
+            return new CustomTrackListener();
+        }
+
+        @Override
+        protected void scrollDueToClickInTrack(int direction) {
+            try {
+                //Set the thumb position to the mouse press location, as opposed
+                //to the closest "block" which is the default behavior.
+                Point mousePosition = slider.getMousePosition();
+                if (mousePosition == null) {
+                    return;
+                }
+                int value = this.valueForXPosition(mousePosition.x);
+
+                //Lock the slider down, which is a shared resource.
+                //The VideoPanelUpdater (dedicated thread) keeps the
+                //slider in sync with the video position, so without 
+                //proper locking our change could be overwritten.
+                sliderLock.acquire();
+                slider.setValueIsAdjusting(true);
+                slider.setValue(value);
+                slider.setValueIsAdjusting(false);
+                sliderLock.release();
+            } catch (InterruptedException ex) {
+                //Thread (EDT) interrupted while waiting on lock.
```

Comment:
will the lock be released if there was an exception?

Reply:
Since I don't do anything in these blocks, I should instead use acquireUninterruptibly()